### ISSUE TITLE
24 words Ethereum seed, performance issues / bad tokenlist ?

### ISSUE THREAD

**Issue by soragXYZ** [3rdIteration/btcrecover#388]
Hi!

I'm trying to bruteforce a 24 words Ethereum seed.

Here is the command I use:
`python seedrecover.py --no-dupchecks --mnemonic-length 24 --language EN --dsw --wallet-type ethereum --addr-limit 1 --addressdb myDB.db --tokenlist testSeed.txt --bip32-path "m/44'/60'/0'" --no-eta`

Here is an example of my tokenlist:
```
+ ^1^STH
+ ^2^STH
+ ^3^STH
+ ^4^STH
+ ^5^STH
+ ^6^STH
+ ^7^STH

+ ^9^STH


+ ^12^STH
+ ^13^STH
+ ^14^STH
+ ^15^STH
+ ^16^STH


+ ^19^STH
+ ^20^STH
+ ^21^STH
+ ^22^STH
+ ^23^STH


+ ^8^A ^10^B ^10^C ^11^D ^17^E ^18^F ^24^G
+ ^8^A ^10^B ^10^C ^11^D ^17^E ^18^F ^24^G
+ ^8^A ^10^B ^10^C ^11^D ^17^E ^18^F ^24^G
+ ^8^A ^10^B ^10^C ^11^D ^17^E ^18^F ^24^G

+ abandon ability able ...
+ abandon ability able ...
```

As you already probably guessed with this tokenlist, I want to try every combination with 2 missing words, but only on position 8, 10, 11, 17, 18 and 24.
And my best guess for word 8 is A, word 10 is B and C, word 11 is D etc... And I don't know the adress I'm looking for, which is why I'm using a database

In other words, I think that I have 2 mistakes, either in position 8, 10, 11, 17, 18 or 24

But for some reason, this bruteforce is like super slow, 1kP/s. I'm usually at 100kP/s with other conditions (no GPU acceleration for both).

Is my tokenlist not optimal ?

**Comment by soragXYZ** [comment]
Ok, I did some readings, and it seems that the process is slow because the bottleneck is on seed generation (see [this](https://github.com/3rdIteration/btcrecover/issues/37#issuecomment-647187454) and [this](https://github.com/3rdIteration/btcrecover/issues/93#issuecomment-760338957)), and not resolution

So is it better to use the command with 2 big typo and "manually" doing the permutations ? ie using this command:

`python seedrecover.py --no-dupchecks --mnemonic-length 24 --language EN --dsw --wallet-type ethereum --addr-limit 1 --addressdb myDB.db --bip32-path "m/44'/60'/0'" --typo 2 --big-typos 2 --no-eta`

then mannualy doing the permutations, ie

```
1 - STH
2 - STH
3 - STH
4 - STH
5 - STH
6 - STH
7 - STH
8 - UNKNOWN
9 - STH
10 - UNKNOWN
11 - STH
12 - STH
13 - STH
14 - STH
15 - STH
16 - STH
17 - STH
18 - STH
19 - STH
20 - STH
21 - STH
22 - STH
23 - STH
24 - STH
```

Then
```
1 - STH
2 - STH
3 - STH
4 - STH
5 - STH
6 - STH
7 - STH
8 - UNKNOWN
9 - STH
10 - STH
11 - UNKNOWN
12 - STH
13 - STH
14 - STH
15 - STH
16 - STH
17 - STH
18 - STH
19 - STH
20 - STH
21 - STH
22 - STH
23 - STH
24 - STH
```

```
1 - STH
2 - STH
3 - STH
4 - STH
5 - STH
6 - STH
7 - STH
8 - UNKNOWN
9 - STH
10 - STH
11 - STH
12 - STH
13 - STH
14 - STH
15 - STH
16 - STH
17 - UNKNOWN
18 - STH
19 - STH
20 - STH
21 - STH
22 - STH
23 - STH
24 - STH
```

And so on

**Comment by 3rdIteration** [comment]
This isn't a support channel, ask I'm the YouTube comments.
